### PR TITLE
Add JOSS paper

### PR DESCRIPTION
**Pull request by hajdik**
## Purpose
Bring the paper branch into main as the [review](https://github.com/openjournals/joss-reviews/issues/5319) is nearly complete. Also bump the version and add a metadata file for the zenodo build to pick up on.

## Expected time until merged
A day or two so we can do a final proofread

## Type of change
- [ ] Bugfix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (non-backwards-compatible fix or feature)
- [ ] Code style update (formatting, renaming)
- [ ] Refactoring (no functional changes, no API changes)
- [ ] Documentation update
- [ ] Maintenance update
- [x] Other (final paper!)

## Testing
PDF can be viewed [here](https://github.com/mdolab/pygeo/actions/workflows/draft-pdf.yml)

## Checklist
- [x] I have run `flake8` and `black` to make sure the Python code adheres to PEP-8 and is consistently formatted
- [ ] I have formatted the Fortran code with `fprettify` or C/C++ code with `clang-format` as applicable
- [ ] I have run unit and regression tests which pass locally with my changes
- [ ] I have added new tests that prove my fix is effective or that my feature works
- [ ] I have added necessary documentation

### CODE REVIEW

File: .github/workflows/draft-pdf.yml [hunk 1]
```diff
@@ -0,0 +1,26 @@
+on:
```
Comment (sseraj) [inline .github/workflows/draft-pdf.yml:1]: This file should be removed before merging

File: paper/paper.md [hunk 2]
```diff
@@ -0,0 +1,201 @@
...
+Both MACH and MPhys use pyOptSparse [@Wu2020a] to interface with optimization algorithms.
+
+[^1]: \url{https://github.com/OpenMDAO/mphys}
+
+pyGeo's interface for design variables and constraints is independent of which solvers are accessing the geometry.
+This means that pyGeo geometries can interact with different types of solvers, such as structures and aerodynamics, in the same way.
+This also allows direct comparison of the behavior or performance of two different solvers within the same discipline using the same geometry parameterization for each, such as two different flow solvers [@Adler2022c].
+
+## Geometry Parameterization with pyGeo
+pyGeo contains several options for parameterizing geometry: variations on the FFD method, interfaces to external parametric modeling tools, and an analytic parameterization.
```
Comment (sseraj) [inline paper/paper.md:74]: The FFD acronym is used before it is defined

File: paper/paper.md [hunk 3]
```diff
@@ -0,0 +1,201 @@
...
+ESP is a CAD software, while OpenVSP is a conceptual design tool.
+The two packages have different capabilities, but both directly define the geometry with design variables, and the created geometry can be used in external analysis tools.
+
+pyGeo interfaces with ESP and OpenVSP in similar ways.
+In both cases, pyGeo takes an instance of the model and its points are associated with coordinates in a mesh from a solver in the MACH framework.
+For ESP (\autoref{fig:esp_example}) and OpenVSP models (\autoref{fig:vsp_example}), the pyGeo interface to the respective software stores the model in a form usable within the MACH framework and updates it as design variables are changed throughout the optimization.
+
+![Example of ESP models of hydrogen tanks used through pyGeo [@Brelje2021a]. \label{fig:esp_example}](esp_example.png)
+
+![Example of a VSP aircraft model used through VSP's pyGeo interface [@Yildirim2022a]. \label{fig:vsp_example}](vsp_example.png)
```
Comment (sseraj) [inline paper/paper.md:125]: The caption should say OpenVSP instead of VSP for consistency with the text

File: paper/paper.md [hunk 4]
```diff
@@ -0,0 +1,201 @@
...
+
+## Constraints
+pyGeo also includes geometric constraints to prevent geometrically undesirable designs.
+The most commonly-used class of geometry constraints in pyGeo involves tracking one or more linear dimensions on the optimized object's surface.
+These constraints are created by specifying a single point, a line, or an array of points, along with a normal direction, then computing two line-surface intersection points.
+Some commonly used geometric constraints in shape optimization, such as thickness, area, and volume constraints (\autoref{fig:constraint}) can be computed using variations on this approach, which is computationally cheap and robust [@Brelje2020a].
+
+![Thickness and volume constraints demonstrated on an wing section [@Brelje2020a]. \label{fig:constraint}](constraints_3d.pdf)
+
+If a more complex geometry needs to be integrated into an optimized surface, pyGeo supports an alternative geometric constraint formulation based on arbitrary triangulated surfaces as described in [@Brelje2020a] and illustrated in \autoref{fig:trisurf}.
```
Comment (sseraj) [inline paper/paper.md:143]: I would change this to:
"... arbitrary triangulated surfaces [@Brelje2020a], illustrated in \autoref{fig:trisurf}."

File: paper/paper.md [hunk 5]
```diff
@@ -0,0 +1,201 @@
...
+where $g$ is the vector of geometric constraints.
+
+For the FFD parameterization, these derivatives are computed using a combination of analytic methods [@Martins2021] and the complex-step method [@Martins2003a].
+For the interfaces to OpenVSP and ESP, the derivatives are computed with parallel finite differences.
+The CST derivatives are computed analytically.
+
+
+# Statement of Need
+Few open-source packages exist with comparable functionalities.
+To the authors' best knowledge, the only other open-source CFD-based optimization framework that contains a geometry parameterization is SU2 [@Economon2016a].
```
Comment (sseraj) [inline paper/paper.md:175]: Is "that contains a geometry parameterization" correct or missing something at the end?

**Review by sseraj — COMMENTED** [review]
I left some minor comments

File: .zenodo.json [hunk 6]
```diff
@@ -0,0 +1,40 @@
+{
+    "creators": [
```
Comment (nwu63) [inline .zenodo.json:2]: Also needs title here

File: paper/paper.md [hunk 7]
```diff
@@ -0,0 +1,201 @@
...
+\end{equation*}
+where $X_s$ is the vector of surface mesh coordinates, and $x$ is the vector of geometric design variables.
+
+Similarly, pyGeo can compute the constraint Jacobian matrix
+\begin{equation*}
+\frac{\mathrm{d}g}{\mathrm{d}x} ,
+\end{equation*}
+where $g$ is the vector of geometric constraints.
+
+For the FFD parameterization, these derivatives are computed using implicit analytic methods (Sec.~6.7)[@Martins2021] and the complex-step method [@Martins2003a].
```
Comment (nwu63) [inline paper/paper.md:168]: The `(Sec.~6.7)` is not rendered correctly.

**Review by sseraj — APPROVED** [review]

File: paper/paper.md [hunk 8]
```diff
@@ -0,0 +1,201 @@
...
+
+![Thickness and volume constraints demonstrated on an wing section [@Brelje2020a]. \label{fig:constraint}](constraints_3d.pdf){ width=75% }
+
+If a more complex geometry needs to be integrated into an optimized surface, pyGeo supports an alternative geometric constraint formulation based on arbitrary triangulated surfaces as illustrated in \autoref{fig:trisurf} [@Brelje2020a].
+
+![Triangulated surface constraint used to optimize an aeroshell around a complex geometry [@Brelje2020a].\label{fig:trisurf}](trisurfcon.pdf)
+
+
+# Parallelism
+pyGeo can optionally work under distributed memory parallelism using MPI, which is required when interfacing with large-scale CFD applications.
```
Comment (nwu63) [inline paper/paper.md:149]: MPI is not defined anywhere

Comment (nwu63) [reply paper/paper.md:149]: Same for CFD.

**Review by nwu63 — COMMENTED** [review]
LGTM other than two minor things. Once fixed (and required approvals are obtained) we can remove the GHA file, and merge.

File: paper/paper.md [hunk 9]
```diff
@@ -0,0 +1,201 @@
...
+\end{equation*}
+where $X_s$ is the vector of surface mesh coordinates, and $x$ is the vector of geometric design variables.
+
+Similarly, pyGeo can compute the constraint Jacobian matrix
+\begin{equation*}
+\frac{\mathrm{d}g}{\mathrm{d}x} ,
+\end{equation*}
+where $g$ is the vector of geometric constraints.
+
+For the FFD parameterization, these derivatives are computed using implicit analytic methods [@Martins2021, Sec. 6.7] and the complex-step method [@Martins2003a].
```
Comment (sseraj) [inline paper/paper.md:168]: One other point that I missed earlier: I don't think "implicit analytic methods" is correct. This section of the book refers to the direct and adjoint methods, but the FFD parameterization does not solve any linear systems. The FFD parameterization is an explicit function that is differentiated mostly by hand (and some complex step).

**Review by nwu63 — APPROVED** [review]
LGTM, don't forget to make a release after merged!

**Pull request merged**